packages/sessions-sdk-react/src/components/session-provider.tsx

```diff
@@ -91,7 +91,7 @@ type Props = ConstrainedOmit<
   privacyPolicyUrl?: string | undefined;
 } & (
     | {
-        paymaster: ComponentProps<typeof SessionProvider>["paymaster"];
+        paymaster?: ComponentProps<typeof SessionProvider>["paymaster"];
         sendToPaymaster?: undefined;
```

Comment:
why did you choose to include `sendToPaymaster`, `sponsor` when `paymaster` is provided and respectively `paymaster` when `sendToPaymaster` is provided

wouldn't 
```ts
{ paymaster?: ComponentProps<typeof SessionProvider>["paymaster"] ;} 
| {
        sendToPaymaster: NonNullable<
          ComponentProps<typeof SessionProvider>["sendToPaymaster"]
        >;
        sponsor: PublicKey | string;
      }
```
make more sense?

Reply:
the argument about not needing awkward code to narrow the type makes sense to me